### PR TITLE
[CARBONDATA-1326] Fixed high priority findbug issues

### PR DESCRIPTION
**Pull request by manishgupta88**
Fixed high priority findbug issues in the code

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/576/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3172/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3173/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/578/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3174/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/579/

File: core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java [hunk 1]
```diff
@@ -69,9 +69,11 @@ public Integer getPageId() {
     if (obj == null || !(obj instanceof DeleteDeltaBlockletDetails)) {
       return false;
     }
-
     DeleteDeltaBlockletDetails that = (DeleteDeltaBlockletDetails) obj;
-    return id.equals(that.id) && pageId == that.pageId;
+    if (null == pageId || null == that.pageId) {
+      return false;
+    }
+    return id.equals(that.id) && pageId.intValue() == that.pageId.intValue();
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java]: change to `return id.equals(that.id) && (pageId.intValue() == that.pageId.intValue());`

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/580/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3175/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3176/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/581/

**Pull request closed without merging**